### PR TITLE
chore[python]: prefer "Sequence" to "list" in various type signatures

### PR DESCRIPTION
**Pull request by alexander-beedie**
Fixes some unnecessarily-restrictive type signatures; quite a few of the `list[x]` types can be replaced with `Sequence[x]` instead (if used for `isinstance` checks you do need to ensure it doesn't inadvertently match `str` though).

Advantage of Sequence vs List is that, in 99% of cases, a `tuple` or other equivalent sequence input is equally valid, and shouldn't result in typing errors/warnings in calling code.

----

_(Arguable if this is better tagged as 'chore', 'refactor', or 'style' - @stinodego, as you're deep into docs/typing and refactoring, do you have any opinion on what fits better? Or even something else? :)_

### CODE REVIEW

File: py-polars/polars/internals/lazy_functions.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def col(
     if isinstance(name, DataType):
         return pli.wrap_expr(_dtype_cols([name]))
 
-    if isinstance(name, list):
+    elif not isinstance(name, str) and isinstance(name, Sequence):
```
[the remark this answers is not in the document]
Comment (alexander-beedie) [reply py-polars/polars/internals/lazy_functions.py:172]: Ahhh, nice; `is_str_sequence` doesn't do quite the same thing, but I should clearly add a new utility function there and replace the ad-hoc checks! (Probably `is_sequence_not_str` would be a suitable name). Will do that now.

[the remark this answers is not in the document]
Comment (alexander-beedie) [reply py-polars/polars/internals/lazy_functions.py:172]: Ended-up simply prioritizing the explicit checks earlier in the if/else, which was clean; while I was looking through the utility options  I did add a new utility type check `is_dtype_sequence` for future use though ;)

**Comment by stinodego** [comment]
> Arguable if this is better tagged as 'chore', 'refactor', or 'style'

I usually refer to [this table](https://github.com/pvdlg/conventional-changelog-metahub#commit-types). So this would classify as a `refactor` 😃

**Comment by ritchie46** [comment]
> > Arguable if this is better tagged as 'chore', 'refactor', or 'style'
> 
> I usually refer to [this table](https://github.com/pvdlg/conventional-changelog-metahub#commit-types). So this would classify as a `refactor` smiley

And what would be the conclusion? I am still not sure.

**Comment by alexander-beedie** [comment]
> I usually refer to [this table](https://github.com/pvdlg/conventional-changelog-metahub#commit-types). 

Nice! I'll bookmark and refer to it for the slightly less obvious cases. (Or could we consider using "typing" for this type of PR? It's not standard, but doesn't seem like a crazy leap - and I did spot that they don't claim the suggested categories are exhaustive... :)

_Other than that, the flexibility of Conventional Commits allows your team to come up with their own types and change those types over time._ [[link](https://www.conventionalcommits.org/en/v1.0.0/#might-conventional-commits-lead-developers-to-limit-the-type-of-commits-they-make-because-theyll-be-thinking-in-the-types-provided)]

**Pull request merged**